src/main/kotlin/sh/hsp/techtree/FileSystemYamlReader.kt

```diff
@@ -1,19 +1,17 @@
 package sh.hsp.techtree
 
+import sh.hsp.techtree.TreeModelValidator.Companion.validate
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
-import com.fasterxml.jackson.module.kotlin.readValue
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import java.io.File
 import java.net.URL
 
-class FileSystemYamlReader : YamlReader {
+class FileSystemYamlReader(private val mapper: ObjectMapper =
+    ObjectMapper(YAMLFactory()).registerKotlinModule()) : YamlReader {
     override fun readToModel(resource: URL): TreeModel {
```

Comment:
Możesz to też zapisać jako:
```
override fun readToModel(resource: URL): TreeModel =
        File(resource.toURI())
            .let { mapper.readValue(it, TreeModel::class.java) }
            .also { validate(it) }
```

Tutaj masz najbezpieczniejszą wersje funkcji - to jest taki najpoprawnieszy idiomatyzm:
https://kotlinlang.org/docs/idioms.html#single-expression-functions

Tutaj można zmienić, ale nie trzeba.